submissions/LuckyDnepr/OOP-Exercise/index.js

```diff
@@ -0,0 +1,105 @@
+"use strict";
+
+class Saying {
+    constructor(person, vocabulary = "vocabulary") {
+        return function (aventForAnswer) {
+            return person[vocabulary].hasOwnProperty(aventForAnswer) ?
+                person[vocabulary][aventForAnswer] :
+                "Nothing to say";
+        };
+    }
+}
+
+class Inhabitant {
+    constructor(species, gender, name) {
+        this.species = species;
+        this.gender = gender;
+        this.name = name;
+    }
+    getInfo() {
+        return `I'm ${this.species}. My name is ${this.name} and I am ${this.gender}.`;
+    }
+}
+
+class Humans extends Inhabitant {
+    constructor(gender, name) {
+        super("Human", gender, name);
+        this.hands = 2;
+        this.legs = 2;
+        this.vocabulary = {
+            "hi": "Hi all!"
+        };
+        this.say = new Saying(this);
+    }
+    getInfo() {
+        return (
+            super.getInfo() + ` I have ${this.hands} hands and ${this.legs} legs.`
+        );
+    }
+}
+
+class Animals extends Inhabitant {
+    constructor(gender, name) {
+        super("Animal", name, gender);
+        this.paws = 4;
+    }
+    getInfo() {
+        return super.getInfo() + ` I have ${this.paws} paws.`;
+    }
+}
+
+class Cat extends Animals {
+    constructor(gender, name) {
+        super(name, gender);
+        this.vocabulary = {
+            "hi": "Nyav nyav!"
+        };
+        this.say = new Saying(this);
+    }
+    getInfo() {
+        return `I'm ${this.constructor.name}. ` + super.getInfo();
```

Comment:
You can use `this.constructor.name` and send all the way to the base class. Class `Animals` will receive it and pas ith thorugh to the Inhabitant. This way you wil avoid confusing output when some inhabitants have something else beside their species.

This looks a bit inconsistently.
![image](https://user-images.githubusercontent.com/8686631/186972396-7d139f41-b9a7-4ca3-b3fd-71168a2c1e79.png)

BTW, by convention, class name should be singular. Look how misguiding `new Animals` may be. Also we can create multiple dogs using class `Dog` but this is not a reason to name class `Dogs`. Plural is used when function/method creates multiple instances - lists, collections etc.